### PR TITLE
[dagster-airbyte] Add optional connection name filter when generating assets

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds the optional `connection_filter` param to both `load_assets_from_airbyte_project` and `load_assets_from_airbyte_instance`.

This user-supplied function will filter down the set of connections that are turned into Dagster assets, in the case that a user only wants to create assets for a subset of connections (e.g. splitting into multiple repos):

```python
ab_assets = load_assets_from_airbyte_project(
    project_dir=MY_AIRBYTE_PROJECT_DIR,
    connection_filter=lambda name: "snowflake" in name,
)

```

## Test Plan

Amended unit tests, tested locally w/ Airbyte.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9937** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9937" target="_blank">Graphite</a> 
    * **PR #9939** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9939" target="_blank">Graphite</a> 
      * **PR #9975** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9975" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9975?utm_source=stack-comment).

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def _clean_name(name: str) -> str:
     return re.sub(r"[^a-z0-9]+", "_", name.lower())
 
 
-class AirbyteConnection(
+class AirbyteConnectionMetadata(
     NamedTuple(
         "_AirbyteConnection",
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py:332]: nit: _AirbyteConnectionMetadata

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py [hunk 2]
```diff
@@ -655,8 +654,8 @@ def load_assets_from_airbyte_project(
         connection_to_group_fn (Optional[Callable[[str], Optional[str]]]): Function which returns an asset
             group name for a given Airbyte connection name. If None, no groups will be created. Defaults
             to a basic sanitization function.
-        connection_filter (Optional[Callable[[str], bool]]): Optional function which takes in a connection
-            name and returns False if the connection should be excluded from the output assets.
+        connection_filter (Optional[Callable[[AirbyteConnectionMetadata], bool]]): Optional function which
+            takes in a connection name and returns False if the connection should be excluded from the output assets.
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py:658]: "takes in an AirbyteConnectionMetadata object"

**Review by OwenKephart — APPROVED** [review]
two small nits, otherwise 👍

**Pull request merged**